library/ns1_zone.py

```diff
@@ -195,43 +245,61 @@ def __init__(self):
         NS1ModuleBase.__init__(self, self.module_arg_spec,
                                supports_check_mode=True,
                                mutually_exclusive=self.mutually_exclusive)
-        self.exec_module()
 
-    def api_params(self):
-        params = dict(
-            (key, self.module.params.get(key))
-            for key in ZONE_KEYS
-            if self.module.params.get(key) is not None
-        )
-        return params
+    def sanitize_params(self, params):
```

Comment:
we've all written dozens of these funcs. this is an improved name, but I really like calling them `prepare_for_api` - I think it's descriptive and has the good connotations from both the old name and your improved name.

Another option i don't mind, that arguably better when it's a consistent pattern is to make the method `serialize(self, deserialized)` and name the return value `serialized` - but that's not a great fit here since a good serializer should return a string. (But i like it as a pattern cause you just put any custom mangling in your serialize/deserialize methods - crispy like the way Go marshal/unmarshal works)

TL;DR - how does `prepare_for_api` hit you :)

Reply:
I'm down with changing this to a more descriptive name, but take a look at [this](https://github.com/ns1/ns1-ansible-modules/blob/396abc160f4473572c9996d2f182ff84e9e711b0/library/ns1_zone.py#L298) to see another way this method is now being used (to normalize params for comparison against a retrieved zone).

Still like `prepare_for_api`, given that this is a more general normalization function now, and not just used right before submitting params to the API?

Reply:
good point, down with the name as you have it since it's not a serialization nor is it strictly for API mangling.